File: app/src/main/java/br/com/pedrosilva/tecnonutri/domain/repositories/ProfileRepository.java
package br.com.pedrosilva.tecnonutri.domain.repositories;

import java.util.List;

import br.com.pedrosilva.tecnonutri.domain.entities.FeedItem;
import br.com.pedrosilva.tecnonutri.domain.entities.Profile;

/**
 * Created by psilva on 3/18/17.
 */

public interface ProfileRepository {
    void get(int userId, ProfileCallback callback);
    void loadMorePosts(int userId, int page, int timestamp, ProfileCallback callback);

    interface ProfileCallback {
        void onSuccess(Profile profile, List<FeedItem> feedItems, int page, int timestamp);
        void onError(Throwable t);
    }
}

File: app/src/main/kotlin/br/com/pedrosilva/tecnonutri/domain/repositories/ProfileRepository.kt
package br.com.pedrosilva.tecnonutri.domain.repositories

import br.com.pedrosilva.tecnonutri.domain.entities.FeedItem
import br.com.pedrosilva.tecnonutri.domain.entities.Profile

interface ProfileRepository {

    fun get(userId: Int, callback: ProfileCallback)

    fun loadMorePosts(
        userId: Int,
        page: Int,
        timestamp: Int,
        callback: ProfileCallback
    )

    interface ProfileCallback {
        fun onSuccess(
            profile: Profile,
            feedItems: List<FeedItem>,
            page: Int,
            timestamp: Int
        )

        fun onError(t: Throwable)
    }
}